src/connector/src/sink/big_query.rs

```diff
@@ -308,66 +335,124 @@ impl BigQuerySinkWriter {
         pk_indices: Vec<usize>,
         is_append_only: bool,
     ) -> Result<Self> {
-        let client = config.common.build_client(&config.aws_auth_props).await?;
+        let client = config
+            .common
+            .build_writer_client(&config.aws_auth_props)
+            .await?;
+        let mut descriptor_proto = build_protobuf_schema(
+            schema
+                .fields()
+                .iter()
+                .map(|f| (f.name.as_str(), &f.data_type)),
+            config.common.table.clone(),
+        )?;
+
+        if !is_append_only {
+            let field = FieldDescriptorProto {
+                name: Some(CHANGE_TYPE.to_string()),
+                number: Some((schema.len() + 1) as i32),
+                r#type: Some(field_descriptor_proto::Type::String.into()),
+                ..Default::default()
+            };
+            descriptor_proto.field.push(field);
+        }
+
+        let descriptor_pool = build_protobuf_descriptor_pool(&descriptor_proto);
+        let message_descriptor = descriptor_pool
+            .get_message_by_name(&config.common.table)
+            .ok_or_else(|| {
+                SinkError::BigQuery(anyhow::anyhow!(
+                    "Can't find message proto {}",
+                    &config.common.table
+                ))
+            })?;
+        let row_encoder = ProtoEncoder::new(
+            schema.clone(),
+            None,
+            message_descriptor.clone(),
+            ProtoHeader::None,
+        )?;
         Ok(Self {
+            write_stream: format!(
+                "projects/{}/datasets/{}/tables/{}/streams/_default",
+                config.common.project, config.common.dataset, config.common.table
+            ),
             config,
-            schema: schema.clone(),
+            schema,
             pk_indices,
             client,
             is_append_only,
-            insert_request: TableDataInsertAllRequest::new(),
-            row_encoder: JsonEncoder::new_with_bigquery(schema, None),
+            row_encoder,
+            message_descriptor,
+            writer_pb_schema: ProtoSchema {
+                proto_descriptor: Some(descriptor_proto),
+            },
         })
     }
 
     async fn append_only(&mut self, chunk: StreamChunk) -> Result<()> {
```

Comment:
Do we still need this special `append_only` method? It is only a subset of `upsert`.

Reply:
One difference is that append-only does not add the CHANGE_TYPE column